### ISSUE TITLE
Reg\Events - Internal User can transfer an operation to another operator

### ISSUE THREAD

**Issue by zoeyli-46** [bcgov/cas-registration#1949]
#### Dependencies

- Blocked by #1432 , #2427 
- Blocking #2447 

#### Description:
As an internal user, I need to be able to transfer an Operation between Operators

#### Acceptance Criteria:
Given that I am on the [BCIERS dashboard](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=7781-6471&t=riHU6HU0lkTtoc8R-4) and click the Transfer an Operation or Facility
OR Given that I am on the [Transfer Grid,](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=23653-2739&t=DXPcYyBQv3MoCu6v-4) and I click Make a transfer
Then I am brought to the form [Figma](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=23653-2740&t=DXPcYyBQv3MoCu6v-4)

Given I have selected "Operation" as my answer to the question "What is being transferred?"
Then only the "Operation" and "Effective date of transfer" fields are displayed in the form

Given that I submitted the form, 
Then I am brought to the transfer complete success page
[Figma](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=23653-2840&t=h6dc4NPJ0ohAfkK5-4)

**Note: this ticket only covers creating a new transfer. Editing the transfer will be covered in a separate ticket.**

#### Development Checklist:

 - [x] apply RBAC to API endpoint(s) for making a transfer so that it's only accessible to "cas_admin" and "cas_analyst" users
 in registration/models/event/transfer_event
- [x] a new record should be created in the transfer_event table for the operation in the Transfer Entity form
- [x] a new record should be created in operation_designated_operator_timelinhe table with the form fields. The "effective date of transfer" in the form should be inserted into the start_date column for the new OperationDesignatedOperatorTimeline record, and it should also be inserted into the end_date columns for the OperationDesignatedOperatorTimeline record of the previous designated operator. The status of the new record should be set to To be transferred. Transferring an operation should not change any data in the operation table => **Sep**: After processing the transfer, we create an entry in the FacilityDesignatedOperationTimeline table. (we don't do it if transfer is in the future)

- [ ] for each facility currently assigned to the operator, a new record should be created in facility_designated_operation_timeline table with the form fields. The "effective date of transfer" in the form should be inserted into the start_date column for the new FacilityDesignatedOperationTimeline record, and it should also be inserted into the end_date column for the FacilityDesignatedOperationTimeline record of the previous designated operation. The status of the new record should be set to To be transferred. => **Sep**: This is not valid
- [x] the dropdown options for "Operation" should only include the operations currently associated with the "Current operator" selection
- [ ] create several mock fixtures for transfers of operations, with effective dates both in the distant future and in the past. Mock fixtures should also include both LFOs and SFOs
- [x] recommended: add explanation in docs/backend folder to explain how transfers work - it's complicated and confusing
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

**Comment by patrickisaac** [comment]
Refinement Notes:
- Dev Checklist still required.

**Comment by patrickisaac** [comment]
Refinement Notes:
- @andrea-williams will be providing the Dev Checklist

**Comment by patrickisaac** [comment]
Refinement Notes:
- @andrea-williams - will refine this to be the operation one

**Comment by Sepehr-Sobhani** [comment]
Why do we need to ask the user to select the current operator if we already know which operator an operation belongs to? 🤔

**Comment by andrea-williams** [comment]
> Why do we need to ask the user to select the current operator if we already know which operator an operation belongs to? 🤔

it's just for internal users to help them find the correct operation. 
Example use cases:
- select an operator first so that there's only 3 operations to choose from, rather than having a dropdown list of every operation in BCIERS.
- if multiple operations have the same name (no constraint against this as far as I can remember), we can help ensure the internal user is selecting the correct operation by first specifying the operator

**Comment by patriciarussellCAS** [comment]
Sorry - I put my review notes in the other ticket, but they apply here as well

![Image](https://github.com/user-attachments/assets/eedb2159-ef6a-4f60-9f8a-7f92447768f8)

**Comment by patriciarussellCAS** [comment]
was able to confirm that this works.
